drivers/android-driver/src/main/java/app/cash/sqldelight/driver/android/AndroidSqliteDriver.kt

```diff
@@ -171,10 +171,10 @@ class AndroidSqliteDriver private constructor(
   override fun <R> executeQuery(
     identifier: Int?,
     sql: String,
-    mapper: (SqlCursor) -> R,
+    mapper: (SqlCursor) -> QueryResult<R>,
     parameters: Int,
     binders: (SqlPreparedStatement.() -> Unit)?,
-  ) = execute(identifier, { AndroidQuery(sql, database, parameters) }, binders) { executeQuery(mapper) }
+  ) = execute(identifier, { AndroidQuery(sql, database, parameters) }, binders) { executeQuery(mapper) }.value
```

Comment:
Could we return: `QueryResult.Value` here? (Only use GitHub Web, so maybe this is already the case).

Reply:
Yeah, I made a mistake here. The `executeQuery()` call in the trailing lambda there expected a `(SqlCursor) -> R` and returned `R` so it was returning a `QueryResult<QueryResult<T>>` which isn't right.

Reply:
I hope we don't need to add transformers like `map` and `flatMap` to fix `QueryResult<QueryResult<T>>`...